cpp/cppsdk/gsdk.cpp

```diff
@@ -435,7 +441,11 @@ namespace Microsoft
 
             const std::unordered_map<std::string, std::string>& GSDK::getConfigSettings()
             {
-                return GSDKInternal::get().m_configSettings;
+				GSDKInternal::get().m_configMutex.lock();
+				std::unordered_map<std::string, std::string> copyConfigSettings = GSDKInternal::get().m_configSettings;
+				GSDKInternal::get().m_configMutex.unlock();
```

Comment:
>GSDKInternal::get().m_configMutex.unlock(); [](start = 4, length = 43)

Same comment about using a RAII style lock as above.

Reply:
Also, if you use a scoped lock, you can just return GSDKInternal::get().m_configSettings, because the compiler will do the copy for you and then unlock.

---
In reply to: [377967598](https://github.com/PlayFab/gsdk/pull/43#discussion_r377967598) [](ancestors = 377967598)